### DISCUSSION TITLE
Is a function that queries data from the database inside the server component file considered to be a server action?

### DISCUSSION THREAD

**Discussion by danielsavinoff** [vercel/next.js#66134] [Help] [answered]
### Summary

I am just confused whether a function inside of the server component file is a server action. And if so, would it be a correct usage considering that a server action is being called always with the method POST?
Eg.
```
const getData = async () => *querying for the data and return it*

export default async function Page() {
 const data = await getData()
 return <List>{data.map(...)}</List>
}```

### Additional information

_No response_

### Example

_No response_

**Comment by icyJoseph** [comment, marked as answer]
Hi,

No. A server action is, a server side defined function, that the client can invoke. 

In your case, you are simply just getting data as part of the SSR run. No special client reference is created for your `getData` function, nor can you pass it, as it is, to client components, not will it ever share its JavaScript with the client.
